marvel-app/coordinator/HomePageCoordinator.swift

```diff
@@ -11,23 +11,26 @@ import UIKit
 class HomePageCoordinator: Coordinator {
     var navigationController: UINavigationController?
     var homePageVC: HomePageViewController
+    let vcContainer: ViewControllersContainer
     
-    init(homePageVC: HomePageViewController){
-        self.homePageVC = homePageVC
+    init(vcContainer: ViewControllersContainer, navigationController: UINavigationController){
+        self.vcContainer = vcContainer
+        self.navigationController = navigationController
+        self.homePageVC = vcContainer.container.resolve(HomePageViewController.self)!
```

Comment:
Its better to keep view controller only in function like start or eventOccured. Remove global homePageVC